lib/ui/chip_bar.dart

```diff
@@ -1,51 +1,41 @@
 import 'package:first_project/enums/chip_type.dart';
-import 'package:first_project/providers/user_provider.dart';
+import 'package:first_project/notifiers/filters_notifier.dart';
 import 'package:first_project/ui/my_chip.dart';
 import 'package:first_project/utilities/app_colors.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 
-class ChipBar extends StatefulWidget {
-  const ChipBar({super.key});
-
-  @override
-  State<ChipBar> createState() => _ChipBarState();
-}
+class ChipBar extends StatelessWidget {
+  static const List<String> chipLabels = [
+    'Offline',
+    'Online',
+    'Parental',
+    'Sick',
+    'Vacation',
+  ];
 
-class _ChipBarState extends State<ChipBar> {
-  final Map<String, bool> chipLabels = {'Offline': false, 'Online': false, 'Parental': false, 'Sick': false, 'Vacation': false};
-
-  void _toggle(String key) {
-    setState(() {
-      chipLabels.forEach((k, value) {
-        if (k == key) {
-          chipLabels[k] = !value;
-          context.read<UserProvider>().toggleFilter(key);
-        }
-      });
-    });
-  }
+  const ChipBar({super.key});
 
   @override
   Widget build(BuildContext context) {
-    final labels = chipLabels.keys.toList();
+    FiltersNotifier filtersNotifier = context.watch<FiltersNotifier>();
     return SizedBox(
       height: 48,
       child: ListView.builder(
           scrollDirection: Axis.horizontal,
           itemCount: chipLabels.length,
           itemBuilder: (context, index) {
-            final labelName = labels[index];
+            final labelName = chipLabels[index];
             return GestureDetector(
               onTap: () {
-                _toggle(chipLabels.keys.toList()[index]);
+                filtersNotifier.toggleFilter(labelName);
               },
               child: Padding(
                 padding: const EdgeInsets.only(right: 8.0),
                 child: MyChip(
                   chipType: ChipType.regular,
                   label: labelName,
-                  color: chipLabels[labelName] ?? false ? AppColors.green : AppColors.backroundColor,
+                  color: filtersNotifier.selectedFilters.contains(labelName) ? AppColors.green : AppColors.backroundColor,
```

Comment:
@KarloK34 Decide the color inside the MyChip widget. Use context.select to decide if the chip is selected